src/core/build_operations.jl

```diff
@@ -0,0 +1,54 @@
+function  _build_canonical(::Type{T},
+                            devices::Dict{Symbol, DeviceModel},
+                            branches::Dict{Symbol, DeviceModel},
+                            services::Dict{Symbol, ServiceModel},
+                            sys::PSY.System,
+                            optimizer::Union{Nothing, JuMP.OptimizerFactory};
+                            kwargs...) where {T <: PM.AbstractPowerFormulation}
+
+    canonical = CanonicalModel(T, sys, optimizer)
+
+    # Build Injection devices
+    for mod in devices
+        @info "Building $(mod[2].device) with $(mod[2].formulation) formulation"
+        _internal_device_constructor!(canonical, mod[2].device, mod[2].formulation, T, sys; kwargs...)
+    end
+
+    # Build Network
+    @info "Building $(T) network formulation"
+    construct_network!(canonical, T, sys; kwargs...)
```

Comment:
This function seems to be outdated. It needs to reference the `ModelReference` ...

Reply:
You are correct, this function needs to be updated.